File: problems/problems_1014/Solution.java
package problems.problems_1014;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int maxScoreSightseeingPair(int[] values) {
        int ans = 0, left = values[0];
        for (int i = 1; i < values.length; i++) {
            ans = Math.max(ans, left + values[i] - i);
            left = Math.max(left, values[i] + i);
        }
        return ans;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] values = jsonArrayToIntArray(inputJsonValues[0]);
        return JSON.toJSON(maxScoreSightseeingPair(values));
    }
}

File: problems/problems_1014/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.maxScoreSightseeingPair(test_input)

    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        ans = left = 0
        for i, val in enumerate(values):
            ans = max(ans, left + val - i)
            left = max(left, val + i)
        return ans
